src/garage/torch/algos/dqn.py

```diff
@@ -188,10 +192,18 @@ def _train_once(self, itr, episodes):
         for _ in range(self._n_train_steps):
             if (self.replay_buffer.n_transitions_stored >=
                     self._min_buffer_size):
-                timesteps = self.replay_buffer.sample_timesteps(
-                    self._buffer_batch_size)
-                qf_loss, y, q = tuple(v.cpu().numpy()
-                                      for v in self._optimize_qf(timesteps))
+                if self._prioritized_replay:
```

Comment:
can you instead make the signature of `ReplayBuffer.sample_timesteps` always return `timesteps, weights, indices`? this way DQN doesn't need to know about the RB implementation

Reply:
The algo needs to be aware of the type of replace buffer used regardless because the priorities of the samples need to be updated via the `update_priorities()`call  if PER is being used. I suppose we could add that to the `ReplayBuffer` class as well and have it not do anything if PER is not being used but that seems like a poor design decision as priorities, weights, and indices are relevant only when using PER. 

Also the algo infers whether PER is being used or not via `isinstance(replay_buffer, PERReplayBuffer)` so the user doesn't have to pass in any extra arguments.

Reply:
Yeah, even if there needs to be some specialized logic, it's still better if the types are consistent. It should be fairly easy to fix the other algorithms.

Reply:
Sure, done!